### PR TITLE
Adding unit test for extending unit parser

### PR DESCRIPTION
**Pull request by jonathanrocher**
@senganal Added a test for your fix in #42 . This extend feature used to work and cannot be broken silently.

### CODE REVIEW

File: scimath/units/tests/test_custom_unit_parser.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+""" Tests for customizing the unit parser.
+"""
+import unittest
+
+from scimath.units.api import unit_parser
+from scimath.units.length import mm
+from scimath.units.unit_parser import UnableToParseUnits
+import sample_units
```
Comment (rkern) [inline scimath/units/tests/test_custom_unit_parser.py]: Implicit relative import.

File: scimath/units/tests/sample_units.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+from scimath.units.length import mm
+
+
+custom_unit = mm
+
+custom_unit_with_label = mm
+custom_unit_with_label.label = "cuwl"
```
Comment (rkern) [inline scimath/units/tests/sample_units.py]: This modifies `mm`

**Comment by senganal** [comment]
@jonathanrocher  The existing test suite will fail if `extend` is broken as it is now being used to initialize the parser.. But yeah, additional explicit tests are good..

**Comment by jonathanrocher** [comment]
@senganal @rkern Comments addressed. I agree with @senganal that the test doesn't increase the coverage. It has more of a documentation purpose.

File: scimath/units/tests/sample_units.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+from scimath.units.length import mm
+from copy import copy
+
+
+custom_unit = copy(mm)
+
+custom_unit_with_label = copy(mm)
+custom_unit_with_label.label = "cuwl"
```
Comment (senganal) [inline scimath/units/tests/sample_units.py]: nit: why do we need two units for the test below ? just `custom_unit.label = 'cuwl'` should be sufficient..

File: scimath/units/tests/test_custom_unit_parser.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    def test_extend_from_module(self):
+        # Custom units are not known initially
+        with self.assertRaises(UnableToParseUnits):
+            unit_parser.parse_unit("cuwl", suppress_unknown=False)
+        with self.assertRaises(UnableToParseUnits):
+            unit_parser.parse_unit("custom_unit", suppress_unknown=False)
+
+        unit_parser.parser.extend(sample_units)
+        self.assertEqual(unit_parser.parse_unit("cuwl"), mm)
```
Comment (senganal) [inline scimath/units/tests/test_custom_unit_parser.py]: nit: i find these tests confusing.. i would just import `custom_unit` and check they are equal..

**Comment by senganal** [comment]
LGTM... @jonathanrocher can this be merged ?

**Comment by senganal** [comment]
Merging as the comments have been addressed..

**Pull request merged**